### PR TITLE
refactor: Unordered mode supports user partial orders

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/core/blocks.py [hunk 1]
```diff
@@ -498,7 +498,7 @@ def to_pandas(
         sampling_method: Optional[str] = None,
         random_state: Optional[int] = None,
         *,
-        ordered: Optional[bool] = None,
+        ordered: bool = True,
```
Comment (tswast) [inline bigframes/core/blocks.py:501]: Does this mean total ordering now? That is to say, can ordered = False support partial ordering?

Let's add this info to the docstring.

File: bigframes/core/compile/api.py [hunk 2]
```diff
@@ -25,38 +25,44 @@
...
-def compile_raw(
-    node: bigframes.core.nodes.BigFrameNode,
-) -> Tuple[str, bigframes.core.ordering.TotalOrdering]:
-    """Compile node into sql that exposes all columns, including hidden ordering-only columns."""
-    ir = _STRICT_COMPILER.compile_ordered_ir(node)
-    sql = ir.raw_sql()
-    ordering_info = ir._ordering
-    assert ir.has_total_order
-    return sql, ordering_info  # type: ignore
+class SqlCompiler:
```
Comment (tswast) [inline bigframes/core/compile/api.py:28]: Nit: `SQLCompiler`

> When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.

https://peps.python.org/pep-0008/#descriptive-naming-styles

**Review by tswast — COMMENTED** [review]
Thanks! Looks good. A few nits.

File: bigframes/core/blocks.py (hunk 1 above)
Comment (TrevorBergeron) [reply bigframes/core/blocks.py:501]: Added docstring and indicate that ordered=True will sort rows, but whether that is total or not depends on session setting.

ordered=False will always be completely unordered.

File: bigframes/core/compile/api.py (hunk 2 above)
Comment (TrevorBergeron) [reply bigframes/core/compile/api.py]: renamed

**Review by tswast — APPROVED** [review]

**Pull request merged**